x/programs/cmd/simulator/src/lib.rs

```diff
@@ -339,17 +309,11 @@ where
         self.responses
             .next()
             .ok_or(StepError::Client(ClientError::Eof))?
-    }
-
-    pub fn run_step<T>(
-        &mut self,
-        caller_key: &str,
-        step: &Step,
-    ) -> Result<PlanResponseTyped<T>, StepError>
-    where
-        T: BorshDeserialize,
-    {
-        self._run_step(caller_key, step)?
+            .and_then(|step| {
+                step.error
+                    .clone()
```

Comment:
Replacing this with a `if let ...` pattern to only move the error when there is one